sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -735,7 +735,20 @@ private void ThrowLockLostException()
                     {
                         if (AmqpMessageConverter.TryGetAmqpObjectFromNetObject(pair.Value, MappingType.ApplicationProperty, out var amqpObject))
                         {
-                            rejected.Error.Info.Add(pair.Key, amqpObject);
+                            // Attempting to set the dead letter reason or description header through the properties
+                            var isReasonHeaderKey = (pair.Key == AmqpMessageConstants.DeadLetterReasonHeader);
+                            var isDescriptionHeaderKey = (pair.Key == AmqpMessageConstants.DeadLetterErrorDescriptionHeader);
+
+                            // General case
+                            var isOtherProperty = !(isReasonHeaderKey || isDescriptionHeaderKey);
+
+                            // Only allow setting of a header property if the header was not passed in through the parameters
```

Comment:
We probably need to detect this and throw an exception if the user tries to set the value in both places. We could consider not throwing if the values in both places are identical. It might be easier to add this logic directly to the new method that adds the ability to specify in both places.